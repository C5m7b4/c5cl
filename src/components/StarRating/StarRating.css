.star-rating-button {
  background-color: transparent;
  border: none;
  outline: none;
  cursor: pointer;
}

.star-rating-button.on {
  color: rgb(229, 233, 17);
}

.star-rating-button.off {
  color: #ccc;
}
