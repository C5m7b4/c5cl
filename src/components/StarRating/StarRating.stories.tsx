import React from 'react';
import { Story, Meta } from '@storybook/react';

import StarRating, { StarRatingProps } from './StarRating';

export default {
  title: 'Widgets/Star Rating',
  component: StarRating,
} as Meta;

const Template: Story<StarRatingProps> = (args) => <StarRating {...args} />;

export const StarRatingExample = Template.bind({});
