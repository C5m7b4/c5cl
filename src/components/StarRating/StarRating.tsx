import React, { useState } from 'react';
import './StarRating.css';

export interface StarRatingProps {
  onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
}

const StarRating = (props: StarRatingProps) => {
  const [rating, setRating] = useState(0);
  const [hover, setHover] = useState(0);

  return (
    <div className="star-rating">
      {[...Array(5)].map((star, index) => {
        return (
          <button
            type="button"
            data-testid={`star-${index}`}
            key={`sr-${index}`}
            className={`star-rating-button ${
              index <= (hover || rating) ? 'on' : 'off'
            }`}
            onClick={() => setRating(index)}
            onMouseEnter={() => setHover(index)}
            onMouseLeave={() => setHover(rating)}
          >
            <span className="star">&#9733;</span>
          </button>
        );
      })}
    </div>
  );
};

export default StarRating;
