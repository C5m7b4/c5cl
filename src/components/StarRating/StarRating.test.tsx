import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import StarRating from './StarRating';

const testfn = jest.fn();

describe('Star Rating', () => {
  test('should render properly', () => {
    const { container } = render(<StarRating onClick={testfn} />);
    expect(container).toMatchSnapshot();
  });
  test('should handle mouse enter', () => {
    const { container, getByTestId } = render(<StarRating onClick={testfn} />);
    fireEvent.mouseEnter(screen.getByTestId('star-1'));
    fireEvent.mouseLeave(screen.getByTestId('star-1'));
  });
  test('should handle click', () => {
    const { container, getByTestId } = render(<StarRating onClick={testfn} />);
    fireEvent.click(screen.getByTestId('star-1'));
  });
});
