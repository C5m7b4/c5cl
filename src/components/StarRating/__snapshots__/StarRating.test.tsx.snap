// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Star Rating should render properly 1`] = `
<div>
  <div
    class="star-rating"
  >
    <button
      class="star-rating-button on"
      data-testid="star-0"
      type="button"
    >
      <span
        class="star"
      >
        ★
      </span>
    </button>
    <button
      class="star-rating-button off"
      data-testid="star-1"
      type="button"
    >
      <span
        class="star"
      >
        ★
      </span>
    </button>
    <button
      class="star-rating-button off"
      data-testid="star-2"
      type="button"
    >
      <span
        class="star"
      >
        ★
      </span>
    </button>
    <button
      class="star-rating-button off"
      data-testid="star-3"
      type="button"
    >
      <span
        class="star"
      >
        ★
      </span>
    </button>
    <button
      class="star-rating-button off"
      data-testid="star-4"
      type="button"
    >
      <span
        class="star"
      >
        ★
      </span>
    </button>
  </div>
</div>
`;
